### PR TITLE
feat: add terminal keepalived

### LINKED ISSUE #1335: Feature: web terminal controller
https://github.com/labring/fist/blob/master/terminal/README.md

https://book.kubebuilder.io/quick-start.html

```
apiVersion: terminal.sealos.io/v1
kind: Terminal
metadata:
  name: aws-infra-demo
spec:
  user: xxx
  token: xxx
  keepalived: 5min
```

Then `kubectl apply -f terminal.yaml` will create a web terminal.

### PR DESCRIPTION
**Pull request by gitccl**
- add default APIServer address for Terminal CRD
- add terminal keepalived

Resolve #1335

### CODE REVIEW

File: controllers/terminal/README.md [hunk 1]
```diff
@@ -27,4 +34,9 @@ TerminalSpec
 ## Usage
 1. run `kubectl apply terminal.yaml`
 2. run `kubectl get service serviceName` to get the nodePort that the service expose. The serviceName is same with the terminal name.
-3. Visit http://ip:port
\ No newline at end of file
+3. Visit http://ip:port
```
Comment (cuisongliu) [inline controllers/terminal/README.md:37]: write to status is better ?

Comment (gitccl) [reply controllers/terminal/README.md:37]: Only write port to status?

Comment (cuisongliu) [reply controllers/terminal/README.md:37]: can write domain dns , ex  name.default.cluster.local:8080    @fanux  Is this necessary here?

Comment (fanux) [reply controllers/terminal/README.md:37]: We will create a applications object later, the UI endpoint can set into applications spec feiled.

File: controllers/terminal/controllers/terminal_controller.go [hunk 2]
```diff
@@ -207,6 +241,33 @@ func (r *TerminalReconciler) syncDeployment(ctx context.Context, terminal *termi
...
+		return r.Update(ctx, terminal)
+	}
+	return nil
+}
+
+// isExpired return true if the terminal has expired
+func isExpired(terminal *terminalv1.Terminal) bool {
+	anno := terminal.ObjectMeta.Annotations
+	timeStr, exist := anno[KeepaliveAnnotation]
+	if !exist {
```
Comment (fanux) [inline controllers/terminal/controllers/terminal_controller.go:256]: If not exist we need set this annotations

Comment (fanux) [reply controllers/terminal/controllers/terminal_controller.go:256]: Or just return true to delete this terminal, must set this annotions when create the terminal.

File: controllers/terminal/README.md (hunk 1 above)
Comment (zzjin) [reply controllers/terminal/README.md:37]: After Disscusion, we could add auto support of host like terminal-UID-Random.cloud.sealos.io into ingress/nginx.

**Comment by cuisongliu** [comment]
add some printers ? printcolumn for crd.   https://kubebuilder.io/reference/generating-crd.html#additional-printer-columns. see it .

**Review by fanux — APPROVED** [review]

**Pull request merged**